### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by oriontvv**
This PR adds dependabot config

### CODE REVIEW

**Comment by cuviper** [comment]
For a library crate like this, I'm not a fan of dependabot-forced cargo updates. End users building an application are free to use whatever semver-compatible versions they like. Is there a way to configure that only for major dependency upgrades?

**Comment by oriontvv** [comment]
Yes, we can specify update-types: major [doc](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups)

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 2
+updates:
+  - package-ecosystem: cargo
+    directory: /
```
Comment (cuviper) [inline .github/dependabot.yml:4]: Will it scan for all `Cargo.toml` in the workspace, or only the root?

**Comment by martin-g** [comment]
All will be scanned

**Comment by cuviper** [comment]
OK, thanks! We'll try it out.

**Pull request merged**